File: 2022/ruby/day01.rb
data = <<~EXAMPLE
1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
EXAMPLE

data = $stdin.read

elves = data.split("\n\n").map { |e| e.lines.sum(&:to_i) }

puts "Part 1: #{elves.max}"
puts "Part 2: #{elves.max(3).sum}"

File: 2022/rust/src/bin/day01.rs
#[allow(dead_code)]
const EXAMPLE: &str = r#"1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
"#;

fn main() {
    // let input = EXAMPLE;
    let input = std::io::read_to_string(std::io::stdin()).unwrap();

    let mut elves: Vec<u32> = input
        .split("\n\n")
        .map(|b| b.lines().fold(0, |sum, s| sum + s.parse::<u32>().unwrap()))
        .collect();
    elves.sort_by(|a, b| b.cmp(a));

    println!("Part 1: {}", elves[0]);
    println!("Part 2: {}", elves.iter().take(3).sum::<u32>());
}
